File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        m={}
        for chr in s:
            if chr not in m:
                m[chr]=1
            else:
                m[chr]+=1
        n={}
        for chr in t:
            if chr not in n:
                n[chr]=1
            else:
                n[chr]+=1
        if m==n:
            return True
        else:
            return False

File: 0242-valid-anagram/0242-valid-anagram.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */
var isAnagram = function(s, t) {
    if (s.length!==t.length){
        return false;
    }
    let m={};
    let n={};

    for (let i=0;i<s.length;i++){
        if (!(s[i] in m)){
            m[s[i]]=1
        }else{
            m[s[i]]++;
        }
    }
    
    for (let i=0;i<t.length;i++){
        if (!(t[i] in m) || m[t[i]]===0){
            return false;
        }
            m[t[i]]--;
        }
        return true;
    


};